src/MyApplication.cpp

```diff
@@ -1,67 +1,109 @@
-#include <Corrade/Containers/Optional.h>
-/* Magnum GL headers must always be included before Qt ones */
-#include <Magnum/GL/Framebuffer.h>
+
 #include <Magnum/Platform/GLContext.h>
-#include <QtWidgets/QApplication>
-#include <QtWidgets/QOpenGLWidget>
+#include <Magnum/GL/Framebuffer.h>
+#include <Magnum/GL/Renderer.h>
+#include <Magnum/Math/Color.h>
+
+#include <gtkmm/application.h>
+#include <gtkmm/window.h>
+#include <gtkmm/glarea.h>
 
 using namespace Magnum;
 
-class MyApplication: public QOpenGLWidget {
-    public:
-        explicit MyApplication(Platform::GLContext& context, QWidget* parent = nullptr, Qt::WindowFlags f = nullptr);
+class MagnumWidget : public Gtk::GLArea {
+public:
+    explicit MagnumWidget(Platform::GLContext &context);
 
-    private:
-        void initializeGL() override;
-        void paintGL() override;
+private:
+    void onRealize();
+    bool onRender(const Glib::RefPtr<Gdk::GLContext> &context);
+    void onResize(int width, int height);
+    void onUnrealize();
 
-        Platform::GLContext& _context;
+    Platform::GLContext &_context;
 };
 
-MyApplication::MyApplication(Platform::GLContext& context, QWidget* parent, Qt::WindowFlags f): QOpenGLWidget{parent, f}, _context(context) {
-    /* TODO: Add your context format setup code here */
+MagnumWidget::MagnumWidget(Platform::GLContext &context) : Gtk::GLArea(), _context(context) {
```

Comment:
```suggestion
MagnumWidget::MagnumWidget(Platform::GLContext& context): Gtk::GLArea{}, _context{context} {
```

In particular, `&` (and `*`) goes next to the type, no space in front of `:`, prefer using `{}` over `()` for constructors. Same elsewhere.

Btw., is the explicit call to `Gtk::GLArea` default constructor needed? I think not.